paddle/fluid/operators/interpolate_op.cu

```diff
@@ -91,15 +102,23 @@ __global__ void KeBilinearInterpFw(
     int channel_id = out_id_w / out_img_size;
 
     int out_img_idy = (out_id_w % out_img_size) / out_img_w;
-    int in_img_idy = ratio_h * out_img_idy;
+    int in_img_idy = (align_mode == 0 && !align_corners)
+                         ? static_cast<int>(ratio_h * (out_img_idy + 0.5) - 0.5)
+                         : static_cast<int>(ratio_h * out_img_idy);
```

Comment:
add 
```suggestion
       in_img_idy = max(0, in_img_idy) 
```
also in other calculations like ratio_h * (out_img_idy + 0.5) - 0.5

Reply:
done